libmscore/system.cpp

```diff
@@ -226,49 +220,8 @@ void System::layoutSystem(qreal xo1)
                   for (auto bi : s->brackets()) {
                         if (bi->column() != i || bi->bracketType() == BracketType::NO_BRACKET)
                               continue;
-                        int firstStaff = staffIdx;
-                        int lastStaff  = staffIdx + bi->bracketSpan() - 1;
-                        if (lastStaff >= nstaves)
-                              lastStaff = nstaves - 1;
-
-                        for (; firstStaff <= lastStaff; ++firstStaff) {
-                              if (score()->staff(firstStaff)->show())
-                                    break;
-                              }
-                        for (; lastStaff >= firstStaff; --lastStaff) {
-                              if (score()->staff(lastStaff)->show())
-                                    break;
-                              }
-                        int span = lastStaff - firstStaff + 1;
-                        //
-                        // do not show bracket if it only spans one
-                        // system due to some invisible staves
-                        //
-                        if ((span > 1) || (bi->bracketSpan() == span)) {
-                              //
-                              // this bracket is visible
-                              //
-                              Bracket* b = 0;
-                              int track = staffIdx * VOICES;
-                              for (int k = 0; k < bl.size(); ++k) {
-                                    if (bl[k]->track() == track && bl[k]->column() == i && bl[k]->bracketType() == bi->bracketType()) {
-                                          b = bl.takeAt(k);
-                                          break;
-                                          }
-                                    }
-                              if (b == 0) {
-                                    b = new Bracket(score());
-                                    b->setBracketItem(bi);
-                                    b->setGenerated(true);
-                                    b->setTrack(track);
-                                    }
-                              add(b);
-//                              if (bi->selected() != b->selected()) {
-//                                    bi->selected() ? score()->select(b) : score()->deselect(b);
-//                                    }
-                              b->setStaffSpan(firstStaff, lastStaff);
-                              bracketWidth[i] = qMax(bracketWidth[i], b->width());
-                              }
+                        Bracket* b = createBracket(bi, i, staffIdx, bl, nullptr);
```

Comment:
Maybe passing the first system's measure to `createBracket()` here can prevent creation of extra brackets I described above?